dds/DCPS/RemoveAssociationSweeper.h

```diff
@@ -120,6 +121,23 @@ RemoveAssociationSweeper<T>::cancel_timer(OpenDDS::DCPS::RcHandle<OpenDDS::DCPS:
   return execute_or_enqueue(new CancelCommand(this, info));
 }
 
+template <typename T>
+void
+RemoveAssociationSweeper<T>::cancel_timer(OpenDDS::DCPS::PublicationId writer_id)
+{
+  struct Predicate {
+    Predicate(const OpenDDS::DCPS::PublicationId writer_id) : writer_id_(writer_id) {}
+    OpenDDS::DCPS::PublicationId writer_id_;
+    bool operator() (const RcHandle<WriterInfo>& info) const {
+      return writer_id_ == info->writer_id_;
+    }
+  };
+  OPENDDS_VECTOR(RcHandle<WriterInfo>)::iterator pos = std::find_if(info_set_.begin(), info_set_.end(), Predicate(writer_id));
```

Comment:
This will fail pre-C++11 (I think) due to instantiating a template (`find_if`) with a local class argument.